app/src/main/java/org/mozilla/focus/settings/SettingsFragment.kt

```diff
@@ -40,60 +38,59 @@ class SettingsFragment : PreferenceFragment(), OnSharedPreferenceChangeListener
+            }
+            resources.getString(R.string.pref_key_share_with_friends) -> {
+                if (!debuggingFirebase()) {
+                    createShareAppDialog(preference.context).show()
+                }
+            }
+            resources.getString(R.string.pref_key_about) -> {
+                val intent = InfoActivity.getAboutIntent(preference.context)
+                startActivity(intent)
+            }
+            resources.getString(R.string.pref_key_night_mode_brightness) -> {
+                Settings.getInstance(preference.context).setNightModeSpotlight(true)
+                startActivity(getStartIntentFromSetting(preference.context))
+            }
+            resources.getString(R.string.pref_key_default_browser) -> {
+                TelemetryWrapper.clickDefaultBrowserInSetting()
+            }
+            resources.getString(R.string.pref_key_private_mode_shortcut) -> {
+                TelemetryWrapper.clickPrivateShortcutItemInSettings()
+                createShortcut(preference.context.applicationContext)
+            }
+            resources.getString(R.string.pref_key_debug_page) -> {
+                startActivity(getStartIntent(preference.context))
             }
-        } else if (keyClicked == resources.getString(R.string.pref_key_about)) {
-            val intent = InfoActivity.getAboutIntent(preference.context)
-            startActivity(intent)
-        } else if (keyClicked == resources.getString(R.string.pref_key_night_mode_brightness)) {
-            Settings.getInstance(preference.context).setNightModeSpotlight(true)
-            startActivity(getStartIntentFromSetting(preference.context))
-        } else if (keyClicked == resources.getString(R.string.pref_key_default_browser)) {
-            clickDefaultBrowserInSetting()
-        } else if (keyClicked == resources.getString(R.string.pref_key_private_mode_shortcut)) {
-            clickPrivateShortcutItemInSettings()
-            createShortcut(preference.context.applicationContext)
-        } else if (keyClicked == resources.getString(R.string.pref_key_debug_page)) {
-            startActivity(getStartIntent(preference.context))
         }
         return super.onPreferenceTreeClick(preferenceScreen, preference)
     }
 
-    private fun debugingFirebase(): Boolean {
+    private fun debuggingFirebase(): Boolean {
         debugClicks++
         if (debugClicks > DEBUG_CLICKS_THRESHOLD) {
             val debugShare = Intent()
             debugShare.action = Intent.ACTION_SEND
             debugShare.type = "text/plain"
-            var testingId: String? = ""
-            val firebase = getFirebase()
-            if (firebase != null) {
-                testingId += """
-                    ${firebase.getFcmToken()}
-
-
-                    """.trimIndent()
-            }
-            testingId += TelemetryHolder.get().clientId
+            val testingId = "${getFirebase().getFcmToken()}\n\n${TelemetryHolder.get().clientId}"
```

Comment:
getFirebase() could be null?

Reply:
`getFirebase()` is returned non-null `firebaseContract` object which could be `FirebaseImp` or `FirebaseNoOpImp`.